compiler/runtime/OMRRuntimeAssumptions.hpp

```diff
@@ -268,7 +268,13 @@ class SentinelRuntimeAssumption : public OMR::RuntimeAssumption
 
    virtual uint8_t *getFirstAssumingPC() { return NULL; }
    virtual uint8_t *getLastAssumingPC() { return NULL; }
-   virtual void     dumpInfo() {};
+   virtual void     dumpInfo() {}
+
+   void *getOwningMetadata()                     { return _owningMetaData; }
+   void  setOwningMetadata(void *owningMetadata) { _owningMetaData = owningMetadata; }
+
+   private:
+   void * _owningMetaData;
```

Comment:
I like the comment you have in the commit message for this field.  Why not add it as a code comment here describing the purpose of this field:

```
A pointer to the owning metadata.  This allows one to dangle a chain of runtime assumptions 
associated with a specific compiled body off of a metadata structure that describes said 
body when reifying the assumptions is necessary.
```

Reply:
Done in https://github.com/eclipse/omr/pull/5553/commits/90b975bf9cdcd13acdae2ec7e550e8aa01327b83